code/go/0chain.net/chaincore/transaction/worker.go

```diff
@@ -91,8 +91,45 @@ func RemoveFromPool(ctx context.Context, txns []datastore.Entity) {
 	txn := transactionEntityMetadata.Instance().(*Transaction)
 	collectionName := txn.GetCollectionName()
 
+	logging.Logger.Debug("cleaning invalid transactions")
+	mappedTxns := make(map[string][]*Transaction)
+	for _, e := range txns {
+		blockTx, ok := e.(*Transaction)
+		if !ok {
+			logging.Logger.Error("generate block (invalid entity)", zap.Any("entity", e))
+			continue
+		}
+		mappedTxns[blockTx.GetKey()] = append(mappedTxns[blockTx.GetKey()], blockTx)
```

Comment:
What's the mappedTxns key, seems it's txn hash, but perhaps should be client id.